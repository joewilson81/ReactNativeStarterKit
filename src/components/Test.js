import React, { Component } from 'react';
import { View, Text, Button, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { buttonPush } from '../actions';

export class Test extends Component {
  render() {
    const { containerStyle, welcomeStyle } = styles;

    return (
      <View style={containerStyle}>
        <Text style={welcomeStyle}>
          Welcome to React Native!
        </Text>

        <Button
          title="Trigger the buttonPush action!"
          color="#841584"
          onPress={() => this.props.buttonPush()}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcomeStyle: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  }
});

export default connect(null, { buttonPush })(Test);
