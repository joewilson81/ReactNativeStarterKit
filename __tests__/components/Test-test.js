import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Test } from '../../src/components/Test';

describe('Test Component', () => {
  it('should render self and subcomponents', () => {
    expect(renderer.create(
      <Test
        buttonPush={jest.fn}
      />
    )).toMatchSnapshot();
  });
});
