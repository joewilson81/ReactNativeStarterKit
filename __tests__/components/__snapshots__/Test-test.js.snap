// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should render self and subcomponents 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Trigger the buttonPush action!
      </Text>
    </View>
  </View>
</View>
`;
